### PR TITLE
fix: Dynamically adjust path separator in `conventions.settings.gradle.kts` for OS compatibility

### LINKED ISSUE #212: Gradle Build Failure Due to Project Naming Conventions
Issue:
The project is facing a Gradle build failure due to non-compliance with Gradle's project naming conventions. the error messages indicate issues with the naming conventions in the following files:

\build-logic\build-conventions\build.gradle.kts
\feature\build.gradle.kts
\app\build.gradle.kts
\build-logic\settings-conventions\build.gradle.kts

Error Message:

```groovy
org.gradle.api.InvalidUserDataException: Cannot generate project dependency accessors:
  - Cannot generate project dependency accessors because project '\build-logic\build-conventions\build.gradle.kts' doesn't follow the naming convention: [a-zA-Z]([A-Za-z0-9\-_])*
  - Cannot generate project dependency accessors because project '\feature\build.gradle.kts' doesn't follow the naming convention: [a-zA-Z]([A-Za-z0-9\-_])*
  - Cannot generate project dependency accessors because project '\app\build.gradle.kts' doesn't follow the naming convention: [a-zA-Z]([A-Za-z0-9\-_])*
  - Cannot generate project dependency accessors because project '\build-logic\settings-conventions\build.gradle.kts' doesn't follow the naming convention: [a-zA-Z]([A-Za-z0-9\-_])*
  
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
  ```

Steps to Reproduce:

- Clone the repository.
- Attempt to build the project using Gradle.
- Observe the mentioned error messages.

Additional Information:
Gradle Version: 8.5

### PR DESCRIPTION
**Pull request by hyuwah**
Hi @julioromano, I'm submitting a pull request to address the Gradle build failure on Windows OS. Your review would be much appreciated! Thanks for this cool template!

This commit adjust platform-specific handling of path separators in `conventions.settings.gradle.kts` to use `\` on Windows and `/` on Unix-based OS

fixes #212 

Build on Windows machine

### CODE REVIEW

File: build-logic/settings-conventions/src/main/kotlin/conventions.settings.gradle.kts [hunk 1]
```diff
@@ -1,18 +1,22 @@
+import org.gradle.internal.os.OperatingSystem
+
+val separator = if (OperatingSystem.current().isWindows) '\\' else '/'
```
Comment (julioromano) [inline build-logic/settings-conventions/src/main/kotlin/conventions.settings.gradle.kts:3]:
```suggestion
import java.io.File.separator
```

File: build-logic/settings-conventions/src/main/kotlin/conventions.settings.gradle.kts [hunk 2]
```diff
@@ -1,18 +1,22 @@
+import org.gradle.internal.os.OperatingSystem
```
Comment (julioromano) [inline build-logic/settings-conventions/src/main/kotlin/conventions.settings.gradle.kts:1]:
```suggestion
import org.gradle.internal.os.OperatingSystem
```

File: build-logic/settings-conventions/src/main/kotlin/conventions.settings.gradle.kts [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
+
```
Comment (julioromano) [inline build-logic/settings-conventions/src/main/kotlin/conventions.settings.gradle.kts:2]:
```suggestion
```

**Review by julioromano — CHANGES_REQUESTED** [review]
Thanks for the PR!
It turns out Java already has OS independent path separator APIs: Check out the suggestions.
Then we should be good to go.

**Comment by hyuwah** [comment]
> Thanks for the PR! It turns out Java already has OS independent path separator APIs: Check out the suggestions. Then we should be good to go.

Thanks for the suggestions! I wasn't aware of the existing APIs either. I'll update the PR to use those instead.

**Review by julioromano — COMMENTED** [review]
LGTM
Thanks for your contribution!

**Review by julioromano — APPROVED** [review]

**Pull request merged**